src/components/_checkbox.scss

```diff
@@ -0,0 +1,123 @@
+@use "../design-tokens/color";
+@use "../design-tokens/typography";
+
+$check-box-size: 1rem;
+$check-background: '::before';
+$check-icon: '::after';
+$check-ease-out-exponential: cubic-bezier(0.04, 0.08, 0.5, 0.95);
+
+/* stylelint-disable selector-max-specificity */
+
+@mixin Checkbox {
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  cursor: pointer;
+
+  // ----------
+  //  Elements
+  // ----------
+
+  label {
+    @include typography.Typography__SmallScreenBody;
+
+    position: relative;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    padding: 0;
+    color: currentColor;
+    cursor: pointer;
+
+    @media (min-width: 600px) {
+      @include typography.Typography__Body;
+    }
+
+    &#{$check-background} {
+      order: 0;
+      box-sizing: border-box;
+      width: $check-box-size;
+      height: $check-box-size;
+      margin-right: 0.75rem;
+      background-color: color.$color-white;
+      border: 1px solid color.$color-utilitiesGray;
+      border-radius: 4px;
+      transition-duration: 150ms;
+      transition-property: background-color, border-color;
+      content: '';
+      pointer-events: none;
+    }
+
+    &#{$check-icon} {
+      position: absolute;
+      left: 1px;
+      display: block;
+      width: 14px;
+      height: 14px;
+      background-image: url("data:image/svg+xml,%3Csvg aria-label='check' role='img' viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill='%23FFF' fill-rule='evenodd' clip-rule='evenodd' d='M20.707 5.293a1 1 0 010 1.414l-11 11a1 1 0 01-1.414 0l-5-5a1 1 0 111.414-1.414L9 15.586 19.293 5.293a1 1 0 011.414 0z'/%3E%3C/svg%3E");
```

Comment:
super nit-pick: you can remove the `aria-label` and `role` to shave off a few bytes since we don’t need a11y for images